conf/distro/include/egl.inc

```diff
@@ -6,21 +6,29 @@ SGX_USERLAND_LIBARIES ?= "ti-sgx-ddk-um"
 # and it is safe to use with user space applications linked against Mesa.
 
 def get_egl_handler(d, target):
-    """  Overloading the default EGL/GLES implementation."""
+    """  Overloading the default EGL/GLES/MESA implementation."""
     features = d.getVar('MACHINE_FEATURES', True).split()
 
     mali_libs = d.getVar('MALI_USERLAND_LIBARIES', True);
     sgx_libs = d.getVar('SGX_USERLAND_LIBARIES', True);
 
     if 'mali450' in features:
-        provider = mali_libs.split()[0]
+        if target == 'mesa' or target == 'libgl':
+            provider = "mesa"
```

Comment:
what does this specific case correspond to? when would we use mesa if mali450 is in features?

Reply:
mali450 driver is only binaries (libMali.so and the lib[gbm|EGL|GLES]*.so* symlinks pointing to the former). So the process is essentially building mesa, and replacing the corresponding mesa binaries  with the mali450's ones. So that the include and the pkgconfig files comes from mesa in the mali450 case.

The SGX driver comes with the relevant include and the *.pc files, and mesa-gl is the best fit there.